mailscanner/bin/MailScanner/MessageBatch.pm

```diff
@@ -758,6 +758,28 @@ sub QuarantineInfections {
   }
 }
 
+# Store all the denial-of-service message in the quarantine if they want me to.
+# Quarantine decision has to be done on a per-message basis.
+sub QuarantineDenialOfService {
+  my $this = shift;
+
+  my($id, $message);
+
+  while(($id, $message) = each %{$this->{messages}}) {
+    next unless $message->{denialofservice};
+    next if $message->{quarantinedinfections}; # Optimisation
+    next if MailScanner::Config::Value ('quarantinedenialofservice', $message) !~
+            /1/;
+
+	# force full message store
+	$message->{allreports}{""} .= ' denialofservice';
```

Comment:
This line is used to log the DoS "infection" and to allow store of the full message in `Quarantine::StoreInfections`. I'm not sure if this is the proper way, but for now... it just works :)